src/Sarif/Autogenerated/SarifRewritingVisitor.cs

```diff
@@ -278,13 +236,31 @@ public virtual ExceptionData VisitExceptionData(ExceptionData node)
             return node;
         }
 
-        public virtual ExternalFiles VisitExternalFiles(ExternalFiles node)
+        public virtual ExternalPropertyFile VisitExternalPropertyFile(ExternalPropertyFile node)
+        {
+            if (node != null)
+            {
+                node.FileLocation = VisitNullChecked(node.FileLocation);
+            }
+
+            return node;
+        }
+
+        public virtual ExternalPropertyFiles VisitExternalPropertyFiles(ExternalPropertyFiles node)
         {
             if (node != null)
             {
                 node.Conversion = VisitNullChecked(node.Conversion);
-                node.Files = VisitNullChecked(node.Files);
                 node.Graphs = VisitNullChecked(node.Graphs);
```

Comment:
>Graphs [](start = 21, length = 6)

You need to add a line for `node.Properties`.

Reply:
there is no code emit for visiting properties anywhere currently, this is a TODO if we think it's useful

---
In reply to: [244650481](https://github.com/Microsoft/sarif-sdk/pull/1194#discussion_r244650481) [](ancestors = 244650481)

Reply:
`externalPropertyFiles` has a property `properties`. But unlike every other property named `properties` in the spec, _this_ one is not of type `propertyBag`; it's of type `externalPropertyFile`. So it does have to be visited.

Am I missing something? I wonder if the visitor code gen omits any property whose name happens to be "properties". Checking...

---
In reply to: [244825443](https://github.com/Microsoft/sarif-sdk/pull/1194#discussion_r244825443) [](ancestors = 244825443,244650481)

Reply:
The codegen emits a visit for any object defined in the schema with a `"$ref"` rather than a `"type"`. The `"$ref`" tells the generator that the property's type is defined elsewhere in the schema, so it needs to be visited.

The schema says that `externalPropertyFiles.properties` is `"$ref": "#/definitions/externalPropertyFile"`, so the code generator should have produced a visit. Is it possible you missed it, or intentionally omitted it (because you thought it was a property bag), when you edited the not-yet-autogenerated file?

---
In reply to: [244835332](https://github.com/Microsoft/sarif-sdk/pull/1194#discussion_r244835332) [](ancestors = 244835332,244825443,244650481)

Reply:
yes, i assumed this was a property bag, in which case there's no visitor bug. now that i've fixed that, yes, there is a visitor bug, thanks, will resolve.

---
In reply to: [244839002](https://github.com/Microsoft/sarif-sdk/pull/1194#discussion_r244839002) [](ancestors = 244839002,244835332,244825443,244650481)